src/App.tsx

```diff
@@ -17,18 +69,25 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                onSelectedFilter={onSelectedFilter}
+                onQuery={onQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {loading && <Loader />}
+              {!loading && (
+                <TodoList visibleTodos={currentTodos} todoOpen={TodoOpen} />
```

Comment:
```suggestion
                <TodoList visibleTodos={currentTodos} selectedTodo={openTodo} selectTodo={setOpenTodo} />
```
better to pass two props instead of some strange object